Json API v2 initial documentation (#830)
digital-asset/docs.daml.com: docs/3.1/docs/json-api/v1/index.rst

The output stream has the same format as the output from the
`Contracts Query Stream`_. We further guarantee that for every ``archived`` event
appearing on the stream there has been a matching ``created`` event
earlier in the stream, except in the case of missing
``contractIdAtOffset`` fields in the case described below.

You may supply optional ``offset`` s for the stream, exactly as with
query streams.  However, you should supply with each ``{templateId,
key}`` pair a ``contractIdAtOffset``, which is the contract ID currently
associated with that pair at the point of the given offset, or ``null``
if no contract ID was associated with the pair at that offset.  For
example, with the above keys, if you had one ``"abc123"`` contract but
no ``"def345"`` contract, you might specify:

.. code-block:: json

    [
        {"templateId": "11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Account:Account", "key": {"_1": "Alice", "_2": "abc123"},
         "contractIdAtOffset": "#1:0"},
        {"templateId": "11c8f3ace75868d28136adc5cfc1de265a9ee5ad73fe8f2db97510e3631096a2:Account:Account", "key": {"_1": "Alice", "_2": "def345"},
         "contractIdAtOffset": null}
    ]

> If every ``contractIdAtOffset`` is specified, as is so in the example
> above, you will not receive any ``archived`` events for contracts
> created before the offset *unless* those contracts are identified in a
> ``contractIdAtOffset``.  By contrast, if any ``contractIdAtOffset`` is
> missing, ``archived`` event filtering will be disabled, and you will
> receive "phantom archives" as with query streams.

Reviewer: ```suggestion
If every ``contractIdAtOffset`` is specified, as in the example
above, you do not receive any ``archived`` events for contracts
created before the offset *unless* those contracts are identified in a
``contractIdAtOffset``.  By contrast, if any ``contractIdAtOffset`` is
missing, ``archived`` event filtering is turned off, and you receive "phantom archives" as with query streams.
```

Healthcheck Endpoints
*********************

The HTTP JSON API provides two healthcheck endpoints for integration
with schedulers like
`Kubernetes <https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/>`_.

Liveness Check
==============

- URL: ``/livez``
- Method: ``GET``

A status code of ``200`` indicates a successful liveness check.

This is an unauthenticated endpoint intended to be used as a liveness
probe.

Readiness Check
===============

- URL: ``/readyz``
- Method: ``GET``

A status code of ``200`` indicates a successful readiness check.

This is an unauthenticated endpoint intended to be used as a readiness

> probe. It validates both the ledger connection as well as the database
> connection.

Reviewer: ```suggestion
probe. It validates both the ledger connection and the database
connection.
```
